[Port] Telecommunications Updated Machine UI

Sarchutar opened the pull request:
## About The Pull Request

Port of: https://github.com/tgstation/tgstation/pull/52357

Makes all telecomms machines use tgui.
I additionally added the list for all text to radio circuits back to the receiver interface and made some small fixes/changes to some things (check commit messages for this).

If there is any follow up PR or anything missing let me know.

<details>
<summary>
Picture
</summary>

![tcomms_tgui](https://user-images.githubusercontent.com/53494785/109327413-9956f480-7858-11eb-92cb-50bbe85ee3f8.png)

</details>

## Why It's Good For The Game

HTML just hurts my eyes and this just looks better.

## Changelog
:cl: WondaMegapon, Sarchutar
add: Telecommunication machines now use a TGUI based interface
del: Removed HTML based telecommunication machine code
balance: Telecommunication machines can have their interfaces viewed without a multitool
server: Telecommunication machines now log changed settings
/:cl:

Sarchutar committed:
initial port

ports the pr

Co-authored-by: WondaMegapon <WondaMegapon@users.noreply.github.com>

Sarchutar committed:
additional changes

-adds support for text to radio circuits back to the tcomms receivers
-fixes a space after the frequency name in filtered frequencies
-adds a color to the add frequency button

Sarchutar committed:
Merge branch 'master' into port-tcomms_tgui

Sarchutar committed:
more changes

-remove remote_control stuff
-remove last Options_Menu() and Options_Topic() since it has no use anymore
-fix Telecomms.js from failing
-remove area from broadcasting signals
-fix double math for freq
-change var/n back to original and move n++
-compile tgui

Sarchutar committed:
Merge branch 'master' into port-tcomms_tgui

Sarchutar committed:
add comments

-adds comments to i++ and newfreq = tempfreq
-moves i++ under the is hidden check

Sarchutar committed:
global variable begone

-removes global remote_control variable since circuits don't use is it anymore and it no use otherwise

Mat05usz commented:
Based

Sarchutar committed:
one log change

-removed "added" infront of removed for one log

francinum commented:
I have no idea why these need to be TGUI, but sure whatever

francinum commented:
At least set the theme to `hackerman`

Sarchutar commented:
> At least set the theme to `hackerman`

Are you sure about that? This just makes it look worse.

![hackerman](https://user-images.githubusercontent.com/53494785/109361066-93790780-7888-11eb-8342-ae90eccc8788.png)

francinum commented:
When the hell did hackerman get beveled buttons...
Yeah in that case the standard theme is fine.

Sarchutar commented:
I will resolve that conflict later.

Sarchutar committed:
Merge branch 'master' into port-tcomms_tgui

Sarchutar committed:
conflict

lets hope this resolves the conflict

Sarchutar commented:
Alright, lets hope i resolved that correctly since i never resolved tgui conflicts before.

Sarchutar committed:
Merge branch 'master' into port-tcomms_tgui

Sarchutar committed:
resolve

park66665 commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,202 @@
+			circuit["name"] = O.name
+			circuit["coords"] = "[O.x], [O.y], [O.z]"
+			circuits += list(circuit)
+	data["circuits"] = circuits
```
I think you can move this portion to `/obj/machinery/telecomms/receiver/add_option()`

park66665 commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,202 @@
-		dat += "<hr>"
+	if(!issilicon(usr))
+		if(!istype(usr.get_active_held_item(), /obj/item/multitool))
+			return
```
```suggestion
	if(!issilicon(operator) && !istype(operator.get_active_held_item(), /obj/item/multitool))
		return
```

Using `usr` (it really is typed as `/mob`) here is mostly okay, but let's be safe and consistent.

(Speaking of consistency, you probably *can* replace `operator` with `usr` if you *really* want.)

park66665 commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,202 @@
+					change_frequency = newfreq
+					. = TRUE
+				else
+					change_frequency = 0
```
```suggestion
					change_frequency = 0
					. = TRUE
```

park66665 commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,202 @@
+			. = TRUE
+
+	add_act(action, params)
+	. = TRUE
```
```suggestion
	if(add_act(action, params))
		. = TRUE
```

park66665 reviewed (commented):
I took a cursory look at the code. Things are generally fine, and the UIs work. (well, it shows up and buttons work)

Sarchutar committed:
code review changes

-move text-to-radio circuit stuff to receiver/add_option()
-change if() in ui_act
-change add_act() to if(add_act())
-add . = TRUE to change_frequency = 0

Co-Authored-By: Xenomedes <park246824@gmail.com>

Sarchutar committed:
Merge branch 'master' into port-tcomms_tgui

Sarchutar committed:
tgui

park66665 reviewed (approved):
All clear. This PR is good to go.
